cuml/random_forest_demo.ipynb

```diff
@@ -0,0 +1,286 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Random Forest and Pickling\n",
+    "The Random Forest algorithm is a classification algorithm which builds several decision trees, and aggregates each of their outputs to make a prediction.\n",
+    "\n",
+    "In this notebook we will train a scikit-learn and a cuML Random Forest Classification model. Then the cuML model is pickled and loaded for prediction. We also compare the results of the scikit-learn, non-pickled and pickled cuML models.\n",
+    "\n",
+    "Note that the underlying algorithm in cuML for tree node splits differs from that used in scikit-learn.\n",
+    "\n",
+    "For information on converting your dataset to cuDF format, refer to the [cuDF documentation](https://rapidsai.github.io/projects/cudf/en/latest/)\n",
+    "\n",
+    "For additional information cuML's random forest model: https://rapidsai.github.io/projects/cuml/en/0.10.0/api.html#random-forest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cudf\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import pickle\n",
+    "\n",
+    "from cuml.ensemble import RandomForestClassifier as curfc\n",
+    "\n",
+    "from sklearn.ensemble import RandomForestClassifier as skrfc\n",
+    "from sklearn.metrics import accuracy_score\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.model_selection import train_test_split"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define Parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_samples = 2**17\n",
+    "n_features = 399\n",
+    "n_info = 300\n",
+    "data_type = np.float32"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Data\n",
+    "\n",
+    "### Host"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "X,y = make_classification(n_samples=n_samples,\n",
+    "                          n_features=n_features,\n",
+    "                          n_informative=n_info,\n",
+    "                          random_state=123, n_classes=2)\n",
+    "\n",
+    "X = pd.DataFrame(X.astype(data_type))\n",
+    "y = pd.Series(y.astype(np.int32))\n",
```

Comment:
Since these are tutorial notebooks, sometimes it's nice to add a little extra detail with comments. I would add an inline comment:

`# cuML RandomForestClassifier requires the targets to be integers`

since I've seen people make this mistake before.